damage.py

```diff
@@ -7,3 +7,139 @@ def damage_ad_armor(attack_ad: float, defense_armor: float):
     """
     return attack_ad * 100 / (100 + defense_armor)
 
+
+def damage_normal_auto_attack_no_crit(attack_base_ad: float, attack_bonus_ad: float, attack_lethality: float,
+                                      attack_level: int, attack_armor_pen: float, attack_bonus_armor_pen: float,
+                                      defense_base_armor: float, defense_bonus_armor: float):
+    """
+    Calculates the output damage of an auto attack that doesn't crit
+    The base armor and bonus armor of the champion being attacked should already take into account the flat or
+    percentage armor reduction resulting from spells like Garen E, Trundle R, Olaf Q, Corki E, or items like Black Cleaver
+    """
+    pre_mitigation_damage = attack_base_ad + attack_bonus_ad
+    defense_armor = defense_base_armor + defense_bonus_armor
+    if defense_armor < 0:
+        return pre_mitigation_damage * (2 - 100 / (100 - defense_armor))
+    else:
+        attack_flat_armor_pen = attack_lethality * (0.6 + 0.4 * attack_level / 18)
+        armor_eq = defense_base_armor * (1 - attack_armor_pen) + defense_bonus_armor * (1 - attack_armor_pen) * (
+                1 - attack_bonus_armor_pen)
+        armor_eq = armor_eq - attack_flat_armor_pen
+        if armor_eq < 0:
+            armor_eq = 0
+        return pre_mitigation_damage * 100 / (100 + armor_eq)
```

Comment:
I see that the last return is the same for all functions and looks like the `damage_ad_armor` function. Maybe we could rename the latter and use it in all the damage functions to avoir repetition

Reply:
Done